### ISSUE TITLE
br close / br delete fails: "cursor must be on a leaf to delete" (frankensqlite B-tree bug)

### ISSUE THREAD

**Issue by Gabko14** [Dicklesworthstone/beads_rust#112]
## Summary

`br close` and `br delete` fail with `"Database error: internal error: cursor must be on a leaf to delete"` on every invocation. The error also affects `br update --status` when changing to a status that triggers a `blocked_issues_cache` rebuild. Operations that don't trigger cache rebuilds (`create`, `update --priority`) work fine.

**Version:** br 0.1.20 (v0.1.20@cd5bc27)
**OS:** macOS 26.4 (Darwin 25.4.0, arm64)

## Reproduction

```bash
# Start fresh — remove existing DB and reimport from JSONL
rm -f .beads/beads.db .beads/beads.db-wal
br sync                    # ✅ imports 63 issues successfully

# These work fine (no cache rebuild triggered):
br create --title="Test" --description="Test" --type=task --priority=4   # ✅ Created winzy.ai-xxx
br update winzy.ai-xxx --priority 3                                       # ✅ Updated

# These all fail (trigger blocked_issues_cache rebuild):
br close winzy.ai-xxx --reason="Test"          # ❌ cursor must be on a leaf to delete
br delete winzy.ai-xxx --reason="Test"         # ❌ cursor must be on a leaf to delete
br update winzy.ai-xxx --status closed         # ❌ cursor must be on a leaf to delete
```

## Full error output

```json
{
  "error": {
    "code": "DATABASE_ERROR",
    "message": "Database error: internal error: cursor must be on a leaf to delete",
    "hint": null,
    "retryable": false,
    "context": null
  }
}
```

## Analysis

The error string `"cursor must be on a leaf to delete"` is present in the `br` binary (confirmed via `strings`), originating from frankensqlite's B-tree cursor implementation. The issue appears to be that during a DELETE operation, the B-tree cursor is positioned on an internal node rather than a leaf node.

The `blocked_issues_cache` rebuild (which runs after status changes) likely triggers the failing DELETE — the import log shows a `"Transitive blocked cache rebuild hit max depth 50"` warning, which may indicate the cache table reaches a state where fsqlite's cursor can't navigate correctly for deletion.

**Key observation:** `br doctor` also reports `ERROR schema.tables: Missing tables: ...` (all tables listed as missing), and standard `sqlite3` cannot read the DB at all (`malformed database schema`). This is consistent with #111 (corrupt page-count header) and suggests broader frankensqlite write correctness issues.

## Workaround

Edit `.beads/issues.jsonl` directly to set `"status":"closed"`, then rebuild:

```bash
rm .beads/beads.db .beads/beads.db-wal
br sync
```

**Comment by madanyang** [comment]
happens also in opensuse tumbleweed

NAME="openSUSE Tumbleweed"
# VERSION="20260226"
ID="opensuse-tumbleweed"
ID_LIKE="opensuse suse"
VERSION_ID="20260226"
PRETTY_NAME="openSUSE Tumbleweed"

**Comment by Gabko14** [comment]
## Additional findings: bug is data-volume and B-tree-position dependent

Same binary v0.1.20, macOS arm64. The bug depends on database size and B-tree structure — it's **not universal**:

- **Project A:** 68 issues (58 open) → 1 out of 58 fails on `br close` / `br update`
- **Project B:** 67 issues (4 open) → 0 out of 4 fail
- **Only ~1 issue fails at a time.** The other 57 succeed.
- **The failing issue rotates.** After successfully updating the broken issue (shifting the B-tree), a *different* issue starts failing instead.
- **Small databases never trigger it.** 4 open issues = zero failures. 60 fresh issues via `br create` = also zero failures.
- **Reimporting doesn't help.** `rm beads.db beads.db-wal && br sync --import-only` reproduces the bug immediately on the fresh DB.

Suggests the bug triggers when the B-tree reaches a certain shape/depth during DELETE operations, not by corruption or specific issue content.

**Comment by madanyang** [comment]
>     * **Small databases never trigger it.** 4 open issues = zero failures. 60 fresh issues via `br create` = also zero failures.
> 
>     * **Reimporting doesn't help.** `rm beads.db beads.db-wal && br sync --import-only` reproduces the bug immediately on the fresh DB.
> 

I have the same thing happening 85 issues clean br init created no issues, migrated from bd with 278 continous problems with database

**Comment by Dicklesworthstone** [comment]
Fixed in 0e4b5df. Bumped fsqlite to 7ca6ff1 which includes the B-tree cursor seek successor positioning fix. Interior-node delete is now failure-atomic, preventing the "cursor must be on a leaf to delete" error during blocked_issues_cache rebuilds.
